pkg/proxy/conntrack/conntrack.go

```diff
@@ -46,59 +68,52 @@ func parametersWithFamily(isIPv6 bool, parameters ...string) []string {
 	return parameters
 }
 
-// ClearEntriesForIP uses the conntrack tool to delete the conntrack entries
-// for the UDP connections specified by the given service IP
-func ClearEntriesForIP(execer exec.Interface, ip string, protocol v1.Protocol) error {
-	parameters := parametersWithFamily(utilnet.IsIPv6String(ip), "-D", "--orig-dst", ip, "-p", protoStr(protocol))
-	err := Exec(execer, parameters...)
-	if err != nil && !strings.Contains(err.Error(), NoConnectionToDelete) {
-		// TODO: Better handling for deletion failure. When failure occur, stale udp connection may not get flushed.
-		// These stale udp connection will keep black hole traffic. Making this a best effort operation for now, since it
-		// is expensive to baby-sit all udp connections to kubernetes services.
-		return fmt.Errorf("error deleting connection tracking state for UDP service IP: %s, error: %v", ip, err)
-	}
-	return nil
-}
-
-// Exec executes the conntrack tool using the given parameters
-func Exec(execer exec.Interface, parameters ...string) error {
-	conntrackPath, err := execer.LookPath("conntrack")
+// exec executes the conntrack tool using the given parameters
+func (ct *execCT) exec(parameters ...string) error {
```

Comment:
maybe we should check that `ct.execer` is not nil and return a more meaningful error about it

Reply:
Why would it be nil? And if it was, that would be a programmer error, not a runtime error, so it's more of a `panic` sitation than `return err`.

Reply:
that is true

Reply:
https://blog.theodo.com/2022/08/go-nil-interfaces/